pkg/controllers/status/common.go

```diff
@@ -69,13 +71,24 @@ func updateResourceStatus(
 		klog.Errorf("Failed to aggregate status for resource(%s/%s/%s, Error: %v", gvr, resource.GetNamespace(), resource.GetName(), err)
 		return err
 	}
-	if reflect.DeepEqual(resource, newObj) {
+
+	oldStatus, _, _ := unstructured.NestedFieldNoCopy(resource.Object, "status")
+	newStatus, _, _ := unstructured.NestedFieldNoCopy(newObj.Object, "status")
+	if reflect.DeepEqual(oldStatus, newStatus) {
 		klog.V(3).Infof("Ignore update resource(%s/%s/%s) status as up to date.", gvr, resource.GetNamespace(), resource.GetName())
 		return nil
 	}
 
-	if _, err = dynamicClient.Resource(gvr).Namespace(resource.GetNamespace()).UpdateStatus(context.TODO(), newObj, metav1.UpdateOptions{}); err != nil {
-		klog.Errorf("Failed to update resource(%s/%s/%s), Error: %v", gvr, resource.GetNamespace(), resource.GetName(), err)
+	patchBytes, err := helper.GenReplaceFieldJSONPatch("/status", oldStatus, newStatus)
```

Comment:
It will return ```[]byte(`[]`), nil```, ```[]byte(`[]`)``` representing an empty jsonpatch, which will be safely passed to apiserver when patching, the objects we patch will always `no change`. You can test it with kubectl.

Reply:
Saw that `GenMergePatch()` returns `nil` if `patchBytes == "{}"`, so I followed this, returned `nil` when no patching is required.

Reply:
Just a question, can we use `retry.RetryOnConflict()` ?

Reply:
I don't see how to prevent empty patches from being sent to apiserver.

Reply:
there's a `DeepEqual()` judge in line 77

Reply:
okay, thanks for reminding me. can you make the patch more precise?
```
status:
  availableReplicas: 1
  conditions:
  - lastTransitionTime: "2023-12-05T09:25:15Z"
    lastUpdateTime: "2023-12-05T09:25:15Z"
    message: Deployment has minimum availability.
    reason: MinimumReplicasAvailable
    status: "True"
    type: Available
  - lastTransitionTime: "2023-11-21T11:43:12Z"
    lastUpdateTime: "2023-12-05T09:25:15Z"
    message: ReplicaSet "demo-57755f7c74" has successfully
      progressed.
    reason: NewReplicaSetAvailable
    status: "True"
    type: Progressing
  observedGeneration: 6
  readyReplicas: 0
  replicas: 1
  updatedReplicas: 1
```
if only the field `readyReplicas` changes from 0 to 1, we'd better use such a patch rather than the entire status:
```
[{
"op":"replace",
"path":"/status/readyReplicas",
"value":1
}]
```

Reply:
I believe implementing such functionality can indeed make the final patch appear very concise in certain situations (as you illustrated). However, it may introduce other issues: if there's a need to update multiple fields, should a separate patch operation be written for each field? If it's necessary to consolidate certain patch operations into one, what is the appropriate timing for the consolidation?

Implementing it this way seems like using JSONPatch to emulate MergePatch. If the goal is simplicity in patch operations, it might prove to be more trouble than it's worth in many scenarios, and the implementation is likely to be less straightforward.

If the concern is that other controllers might modify the status, as mentioned in https://github.com/karmada-io/karmada/issues/4093#issuecomment-1838828261, I believe this possibility is low. In the current implementation, `.status` is returned by the interpreterWebhook, and users have the opportunity to introduce other mechanisms for custom handling.